### PR TITLE
Build: Add Sonar PR decoration

### PR DESCRIPTION
**Pull request by xC0dex**
- Remove CodeCov.
- Add Upload/Download coverage reports.
- Use `opencover` format for SonarCloud.
- Explicitly downgrade SonarScanner to version `8.0.1` due to [this issue](https://community.sonarsource.com/t/sonarscanner-for-net-8-now-with-other-languages-auto-detection/120563/18?u=xc0dex).

Needs to be merged before https://github.com/MudBlazor/MudBlazor/pull/9774.

### CODE REVIEW

**Comment by mikes-gh** [comment]
Need to think about the coverage format.  I think this will break ReportGenerator coverage report.  Reportgenerator does support opencover I believe.  Need to make sure local coverage tools still work.  Also we may be able to remove the coverlet package in the main repo

File: .github/workflows/template-build-test.yml [hunk 1]
```diff
@@ -86,12 +81,11 @@ jobs:
         run: dotnet build -c Release --no-restore
 
       - name: Test
-        run: dotnet test -c Release --no-build --no-restore --blame-hang --blame-hang-timeout 60s /property:GenerateFullPaths=true /consoleloggerparameters:NoSummary /p:CollectCoverage=true /p:CoverletOutputFormat=cobertura /p:ExcludeByAttribute='ExcludeFromCodeCoverageAttribute' /p:ExcludeByAttribute='CompilerGeneratedAttribute' /p:ExcludeByFile='**/*.g.cs' /p:Include='[MudBlazor]*' /p:SkipAutoProps=true /p:CoverletOutput='./TestResults/'
```
Comment (mikes-gh) [inline .github/workflows/template-build-test.yml:89]: Just wondering why you removed the include.  Doesnt this now show code coverage outside the library

Comment (xC0dex) [reply .github/workflows/template-build-test.yml:89]: Yeah I think this is required for the Source Generator and the Analyser. Both have tests but the coverage was not included.

Comment (mikes-gh) [reply .github/workflows/template-build-test.yml:89]: How is the coverage excluded from the other projects are they annotated?

**Pull request closed without merging**